src/js/rtc_session.js

```diff
@@ -422,11 +424,25 @@ export class TalkingState extends RTCSessionState {
         this.transit(new DisconnectedState(this._rtcSession));
     }
     onIceStateChange(evt) {
-        if (evt.currentTarget.iceConnectionState == 'disconnected') {
+        this.logger.info('ICE Connection State: ', evt.currentTarget.iceConnectionState);
+
+        if (evt.currentTarget.iceConnectionState == ICE_CONNECTION_STATE.DISCONNECTED) {
             this.logger.info('Lost ICE connection');
             this._rtcSession._sessionReport.iceConnectionsLost += 1;
         }
+        if (evt.currentTarget.iceConnectionState == ICE_CONNECTION_STATE.FAILED) {
+            this._rtcSession._sessionReport.iceConnectionsFailed = true;
+        }
     }
+
+    onPeerConnectionStateChange() {
+        this.logger.info('Peer Connection State: ', this._rtcSession._pc.connectionState);
+
+        if (this._rtcSession._pc.connectionState == PEER_CONNECTION_STATE.FAILED) {
+            this._rtcSession._sessionReport.peerConnectionFailed = true;
```

Comment:
Nit: I see you have getter and setter method defined below. Why are we then assigning these values directly instead of using those method?

Reply:
see setter for example: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/set
when we use report.peerConnectionFailed = true, it is actually calling the setter.
compare to report._peerConnectionFailed = true, that is assigning the values directly. 

I think here we are using the setter and getters, and we just did not have `set` and `get` in the method name